packages/core/src/modules/employment/queries/list-job-offers.ts

```diff
@@ -0,0 +1,73 @@
+import { ListSearchParams } from '@/shared/types';
+import { Company, EmploymentType, JobOfferStatus } from '../employment.types';
+import { db } from '@/infrastructure/database';
+import { sql } from 'kysely';
+import { z } from 'zod';
+import { ExtractValue } from '@oyster/types';
+
+const JobOfferFilterParams = z.object({
+  company: Company.shape.id.nullable().catch(null),
+  employmentType: z.nativeEnum(EmploymentType),
+  location: z.string().trim().min(1).nullable().catch(null),
+  status: z.nativeEnum(JobOfferStatus),
+});
+
+const JobOfferFilterParam = JobOfferFilterParams.keyof().enum;
+
+const JobOfferSearchParams = ListSearchParams.omit({
+  timezone: true,
+  search: true,
+}).merge(JobOfferFilterParams);
+
+type JobOfferFilterParam = ExtractValue<typeof JobOfferFilterParam>;
+type JobOfferSearchParams = z.infer<typeof JobOfferSearchParams>;
+
+async function listJobOffers({
+  company,
+  employmentType,
+  location,
+  status,
+}: JobOfferSearchParams) {
+  const query = db
+    .selectFrom('jobOffers')
+    .$if(!!company, (query) => {
+      return query.where((eb) => {
+        return eb.exists(
+          eb
+            .selectFrom('jobOffers')
+            .where('companies.crunchbaseId', '=', company)
+        );
+      });
+    })
+    .$if(!!employmentType, (query) => {
+      return query.where((eb) => {
+        return eb.exists(
+          eb
+            .selectFom('jobOffers')
+            .where('jobOffers.employmentType', '=', employmentType)
+        );
+      });
+    })
+    .$if(!!location, (query) => {
+      return query.where((eb) => {
+        return eb.exists(
+          eb.selectFrom('jobOffers').where('jobOffers.location', '=', location)
+        );
+      });
+    })
+    .$if(!!status, (query) => {
+      return query.where((eb) => {
+        return eb.exists(
+          eb.selectFrom('jobOffers').where('jobOffers.status', '=', status)
+        );
+      });
+    });
+
+  const [joboffers, { count }] = await Promise.all([
+    query.select([
+      'jobOffers.company',
+      'jobOffers.employmentType',
+      'jobOffers.',
+    ]),
+  ]);
```

Comment:
Actually, for now, you can just return an empty selection and when I get to the front-end, I'll fill in the gaps based on what I need!

```suggestion
  const [jobOffers, { count }] = await Promise.all([
    query
      .select(['jobOffers.id'])
      .orderBy('jobOffers.createdAt', 'desc')
      .limit(limit)
      .offset((page - 1) * limit)
      .execute(),
      
     query
      .select((eb) => eb.fn.countAll().as('count'))
      .executeTakeFirstOrThrow(),
  ]);
  
  return {
    jobOffers,
    totalCount: Number(count)
  }
```